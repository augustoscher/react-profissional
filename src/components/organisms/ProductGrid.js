import React, { useState } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Grid from '../atoms/Grid';
import Card, { CardBody, CardMedia } from '../atoms/Card';
import Heading from '../atoms/Heading';
import Button from '../atoms/Button';

const Toolbar = styled.div`
  margin-top: 40px;
  text-align: center;
`;

const ProductGrid = ({ products }) => {
  const [showAll, setShowAll] = useState(false);
  const filteredProducts = showAll ? products : products.slice(0, 3);

  return (
    <>
      <Grid md={3}>
        {filteredProducts.map(product => (
          <Card key={product.id}>
            <CardMedia image={product.image} />
            <CardBody>
              <Heading>
                <h6>{product.title}</h6>
              </Heading>
              <p>{product.summary}</p>
              <div>
                <Button variant="linked" color="primary">
                  Saiba mais
                </Button>
              </div>
            </CardBody>
          </Card>
        ))}
      </Grid>
      {!showAll && (
        <Toolbar>
          <Button variant="outlined" onClick={() => setShowAll(true)}>
            Lista completa de servi√ßos
          </Button>
        </Toolbar>
      )}
    </>
  );
};

ProductGrid.defaultProps = {
  products: [],
};

ProductGrid.propTypes = {
  products: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
      image: PropTypes.string,
      title: PropTypes.string,
      summary: PropTypes.string,
    })
  ),
};

export default ProductGrid;
