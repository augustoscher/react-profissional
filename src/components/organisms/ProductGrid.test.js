import React from 'react';
import { render, screen } from 'test-utils';

import ProductGrid from './ProductGrid';

const buildProducts = size => {
  return [...Array(size).keys()].map(i => {
    const id = i + 1;
    return {
      id,
      title: `title${id}`,
      image: `image${id}`,
      summary: `summary${id}`,
    };
  });
};

test('match snapshot', () => {
  const { asFragment } = render(<ProductGrid products={buildProducts(6)} />);

  expect(asFragment()).toMatchSnapshot();
});

test.each([
  [1, 1],
  [2, 2],
  [3, 3],
  [4, 3],
  [6, 3],
])('with %i products should render %i product(s)', (size, expected) => {
  render(<ProductGrid products={buildProducts(size)} />);

  expect(screen.getAllByRole('heading').length).toEqual(expected);
});
