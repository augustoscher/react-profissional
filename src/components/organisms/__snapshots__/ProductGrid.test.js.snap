// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-column-gap: 16px;
  grid-row-gap: 16px;
  grid-template-columns: 1fr;
}

.c3 {
  padding: 16px;
}

.c3 h6 {
  margin-top: 0;
}

.c2 {
  background-image: url(image1);
  position: center center;
  background-size: cover;
  height: 270px;
}

.c6 {
  background-image: url(image2);
  position: center center;
  background-size: cover;
  height: 270px;
}

.c7 {
  background-image: url(image3);
  position: center center;
  background-size: cover;
  height: 270px;
}

.c1 {
  background-color: #fff;
  border-radius: 4px;
}

.c4 h1,
.c4 h2,
.c4 h3,
.c4 h4,
.c4 h5,
.c4 h6 {
  position: relative;
  line-height: 1.5;
  margin-bottom: 25px;
  padding-bottom: 8px;
  border-bottom: 1px solid rgba(255,255,255,0.2);
}

.c4 h1:after,
.c4 h2:after,
.c4 h3:after,
.c4 h4:after,
.c4 h5:after,
.c4 h6:after {
  content: '';
  position: absolute;
  left: 0;
  bottom: -3px;
  background-color: #ffc107;
  height: 5px;
  width: 70px;
}

.c4 h1 {
  padding-bottom: 20px;
}

.c5 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #ffc107;
  border: 2px solid #ffc107;
  color: #212121;
  background-color: transparent;
  border-color: transparent;
  color: #ffc107;
  padding-left: 0;
  padding-right: 0;
}

.c5:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c5:hover:enabled {
  background-color: #c79100;
  border-color: #c79100;
}

.c5:hover:enabled {
  background-color: transparent;
  border-color: transparent;
  color: #c79100;
}

@media (min-width:768px) {
  .c0 {
    grid-template-columns: repeat(3,1fr);
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div />
      <div
        class="c2"
      />
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <h6>
            title1
          </h6>
        </div>
        <p>
          summary1
        </p>
        <div>
          <button
            class="c5"
            color="primary"
            type="button"
          >
            Saiba mais
          </button>
        </div>
      </div>
    </div>
    <div
      class="c1"
    >
      <div />
      <div
        class="c6"
      />
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <h6>
            title2
          </h6>
        </div>
        <p>
          summary2
        </p>
        <div>
          <button
            class="c5"
            color="primary"
            type="button"
          >
            Saiba mais
          </button>
        </div>
      </div>
    </div>
    <div
      class="c1"
    >
      <div />
      <div
        class="c7"
      />
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <h6>
            title3
          </h6>
        </div>
        <p>
          summary3
        </p>
        <div>
          <button
            class="c5"
            color="primary"
            type="button"
          >
            Saiba mais
          </button>
        </div>
      </div>
    </div>
  </div>
  .c1 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #e0e0e0;
  border: 2px solid #e0e0e0;
  color: #212121;
  background-color: transparent;
  color: #212121;
}

.c1:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c1:hover:enabled {
  background-color: #5a6268;
  border-color: #5a6268;
}

.c1:hover:enabled {
  background-color: transparent;
  color: #5a6268;
}

.c0 {
  margin-top: 40px;
  text-align: center;
}

@media (min-width:768px) {

}

<div
    class="c0"
  >
    <button
      class="c1"
      color="default"
      type="button"
    >
      Lista completa de servi√ßos
    </button>
  </div>
</DocumentFragment>
`;
